### PR TITLE
Kevin/debug block mining failure

### PR DESCRIPTION
**Pull request by mieubrisse**

### CODE REVIEW

File: kurtosis-module/impl/participant_network/el/mining_waiter/mining_waiter.go [hunk 1]
```diff
@@ -20,6 +21,11 @@ func (waiter *miningWaiter) WaitForMining(numRetries uint32, timeBetweenRetries
 		if err == nil && blockNumber > 0 {
 			return nil
 		}
+		if err != nil {
```
Comment (tedim52) [inline kurtosis-module/impl/participant_network/el/mining_waiter/mining_waiter.go:24]: Just wondering, why not make this an `else if` off the prior conditional?

Comment (tedim52) [inline kurtosis-module/impl/participant_network/el/mining_waiter/mining_waiter.go:24]: Just wondering, why not make this an `else if` off the prior conditional?

Comment (tedim52) [inline kurtosis-module/impl/participant_network/el/mining_waiter/mining_waiter.go:24]: Just wondering, why not make this an `else if` off the prior conditional?

**Review by tedim52 — APPROVED** [review]
approved

**Review by leoporoli — APPROVED** [review]

File: kurtosis-module/impl/participant_network/el/mining_waiter/mining_waiter.go (hunk 1 above)
Comment (mieubrisse) [reply kurtosis-module/impl/participant_network/el/mining_waiter/mining_waiter.go:24]: chatted over Slack; has to do with the "eject early" in this case is actually ejecting on the _success_ case

**Pull request merged**